src/test/java/org/jenkinsci/plugins/docker/workflow/DockerUtilsTest.java

```diff
@@ -106,6 +111,24 @@
         DockerUtils.parseBuildArgs(null, commangLine);
     }
 
+    @Test public void parseBuildArgWithKeyOnlyAndEnvVariable() throws IOException, InterruptedException {
+
+        EnvironmentVariablesNodeProperty prop = new EnvironmentVariablesNodeProperty();
+        EnvVars env = prop.getEnvVars();
+
+        final String json_arg = "EXAMPLE_JSON";
+        final String json_value = "{\"http-basic\":{\"repo.example.org\":{\"username\":\"some-api-user\",\"password\":\"generatedapipasswordorsomeothersecret\"}}}";
+
+        env.put("EXAMPLE_JSON", json_value);
+        j.jenkins.getGlobalNodeProperties().add(prop);
+
+        final String commangLine = "docker build -t hello-world --build-arg "+json_arg;
+        Map<String, String> buildArgs = DockerUtils.parseBuildArgs(null, commangLine, env);
+
+        Assert.assertThat(buildArgs.keySet(), IsCollectionWithSize.hasSize(1));
```

Comment:
There's a specific matcher for maps, no need to extract the keyset
```suggestion
        Assert.assertThat(buildArgs, IsMapWithSize.aMapWithSize(1));
```